node/forest_libp2p/bitswap/src/request_manager.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::{event_handlers::*, *};
+use ahash::{HashMap, HashMapExt, HashSet, HashSetExt};
+use flume::TryRecvError;
+use libipld::Block;
+use libipld::Cid;
+use libp2p::PeerId;
+use parking_lot::RwLock;
+use std::{
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+const BITSWAP_BLOCK_REQUEST_INTERVAL: Duration = Duration::from_millis(500);
+
+#[derive(Debug, Clone)]
+struct ResponseChannels {
+    block_have: flume::Sender<PeerId>,
+    block_saved: flume::Sender<()>,
+}
+
+// TODO: Use message queue like `go-bitswap`
```

Comment:
let's have an issue to track this

Reply:
Tracked in https://github.com/ChainSafe/forest/issues/2461